### ISSUE TITLE
Using equal distribution from each folder while training 

### ISSUE THREAD

**Issue by pathikg** [ultralytics/yolov5#10829]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

In my use case, I've documents from multiple organizations each of them having their own folders with images and labels stored in yolo format.
e.g. 
```
├── dataset
│   ├── org1
│        ├── images
│        ├── labels 
│   ├── org2
│        ├── images
│        ├── labels 
 ...
 ```
 
Instead of integrating all of these folders into a single dataset, as I had previously done, I want to train Yolov5 on these documents so that I get an equal distribution of documents from each of the `org` directories, maintaining around equal document distribution for all org while training since there's already a high imbalance between counts of document for each org. 

What modifications must I do in the repository code in order to this? 
OR
Is there any other way to do it without modifying current code base?

**Comment by glenn-jocher** [comment]
@pathikg thanks for reaching out! To achieve an equal distribution of documents from each `org` directory, you can utilize PyTorch's `torch.utils.data.WeightedRandomSampler` to balance the dataset during training. You can find detailed instructions on how to implement this in the Ultralytics YOLOv5 documentation at [https://docs.ultralytics.com/yolov5/](https://docs.ultralytics.com/yolov5/).

Feel free to give it a try and let us know if you encounter any issues. Good luck!
